hack/e2e_test.go

```diff
@@ -0,0 +1,329 @@
+	cases := []struct {
+		args []string
+		expected flags
+	}{
+		{
+			[]string{"hello", "world"},
+			flags{getDefault, oldDefault, []string{"world"}},
+		},
+		{
+			[]string{"hello", "--", "--venus", "--karaoke"},
+			flags{getDefault, oldDefault, []string{"--venus", "--karaoke"}},
+		},
+		{
+			[]string{"hello", "--alpha", "--beta"},
+			flags{getDefault, oldDefault, []string{"--alpha", "--beta"}},
+		},
+		{
+			[]string{"so", "--get", "--boo"},
+			flags{true, oldDefault, []string{"--boo"}},
+		},
+		{
+			[]string{"omg", "--get=false", "--", "ugh"},
+			flags{false, oldDefault, []string{"ugh"}},
+		},
+		{
+			[]string{"wee", "--old=5m", "--get"},
+			flags{true, 5*time.Minute, []string{}},
+		},
+		{
+			[]string{"fun", "--times", "--old=666s"},
+			flags{getDefault, oldDefault, []string{"--times", "--old=666s"}},
+		},
+	}
+
+	for i, c := range cases {
+		a := parse(c.args)
+		e := c.expected
+		if a.get != e.get {
+			t.Errorf("%d: a=%v != e=%v", i, a.get, e.get)
+		}
+		if a.old != e.old {
+			t.Errorf("%d: a=%v != e=%v", i, a.old, e.old)
+		}
+		if !reflect.DeepEqual(a.args, e.args) {
+			t.Errorf("%d: a=%v != e=%v", i, a.args, e.args)
+		}
+	}
+}
+
+func TestLook(t *testing.T) {
+	lpf := errors.New("LookPath failed")
+	sf := errors.New("Stat failed")
+	lpnc := errors.New("LookPath should not be called")
+	snc := errors.New("Stat should not be called")
+	cases := []struct {
+		stat     error
+		lookPath error
+		goPath   string
+		expected error
+	}{
+		{ // GOPATH set, stat succeeds returns gopath
+			nil,
+			lpnc,
+			"fake-gopath/",
+			nil,
+		},
+		{ // GOPATH set, stat fails, terms on lookpath
+			sf,
+			lpf,
+			"fake-gopath/",
+			lpf,
+		},
+		{ // GOPATH unset, stat not called, terms on lookpath
+			snc,
+			lpf,
+			"",
+			lpf,
+		},
+		{ // GOPATH unset, stat not called, lookpath matches
+			snc,
+			nil,
+			"",
+			nil,
+		},
+	}
+
+	for _, c := range cases {
+		l := tester{
+			func(string) (os.FileInfo, error) {
+				return FileInfo{}, c.stat
+			},
+			func(string) (string, error) {
+				if c.lookPath != nil {
+					return "FAILED", c.lookPath
+				}
+				return "$PATH-FOUND", nil
+			},
+			c.goPath,
+			nil, // wait
+		}
+		if _, err := l.look(); err != c.expected {
+			t.Errorf("err: %s != %s", err, c.expected)
+		}
+	}
+}
+
+func TestGetKubetest(t *testing.T) {
+	gp := "fake-gopath"
+	gpk := filepath.Join(gp, "bin", "kubetest")
+	p := "PATH"
+	pk := filepath.Join(p, "kubetest")
+	eu := errors.New("upgrade failed")
+	et := errors.New("touch failed")
+	cases := []struct {
+		get bool
+		old time.Duration
+
+		stat     string        // stat succeeds on this file
+		path     bool          // file exists on path
+		age      time.Duration // age of mod time on file
+		upgraded bool          // go get -u succeeds
+		touched  bool          // touch succeeds
+		goPath   string        // GOPATH var
+
+		returnPath  string
+		returnError error
+	}{
+		{ // 0: Pass when on GOPATH/bin
+			false,
```

Comment:
I prefer explicitly stating the struct members that you're setting in this kind of thing, since it's super hard to read these entries when it's just a list of values.